core/src/components/checkbox/checkbox.scss

```diff
@@ -72,7 +72,9 @@
   flex-grow: 1;
 
   align-items: center;
+  justify-content: space-between;
```

Comment:
This is required so that the user can still do the following, with a space between the label and checkbox:

```html
<ion-checkbox style="width: 100%">Label</ion-checkbox>
```